google/cloud/aiplatform/metadata/resource.py

```diff
@@ -0,0 +1,300 @@
+                project=project, location=location
+            )
+            + f"/metadataStores/{metadata_store_id}"
+        )
+
+        try:
+            cls._create_resource(
+                client=api_client,
+                parent=parent,
+                resource_id=resource_id,
+                schema_title=schema_title,
+                display_name=display_name,
+                schema_version=schema_version,
+                description=description,
+                metadata=metadata,
+            )
+        except exceptions.AlreadyExists:
+            logging.info(f"Resource '{resource_id}' already exist")
+
+        return cls(
+            resource_name=f"{parent}/{cls._resource_noun}/{resource_id}",
+            metadata_store_id=metadata_store_id,
+            project=project,
+            location=location,
+            credentials=credentials,
+        )
+
+    @classmethod
+    def get(
+        cls,
+        resource_name: str,
+        metadata_store_id: Optional[str] = "default",
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        credentials: Optional[auth_credentials.Credentials] = None,
+    ) -> "_Resource":
+        """Returns a metadata Resource.
+
+        Args:
+            resource_name (str):
+                A fully-qualified resource name or resource ID
+                Example: "projects/123/locations/us-central1/metadataStores/default/<resource_noun>/my-resource".
+                or "my-resource" when project and location are initialized or passed.
+            metadata_store_id (str):
+                The metadata_store_id portion of the resource name with
+                the format:
+                projects/123/locations/us-central1/metadataStores/<metadata_store_id>/<resource_noun>/my-resource
+                If not provided, the MetadataStore's ID will be set to "default".
+            project (str):
+                Project to get this resource from. Overrides project set in
+                aiplatform.init.
+            location (str):
+                Location to get this resource from. Overrides location set in
+                aiplatform.init.
+            credentials (auth_credentials.Credentials):
+                Custom credentials to use to get this resource. Overrides
+                credentials set in aiplatform.init.
+
+        Returns:
+            resource (_Resource):
+                Instantiated representation of the managed Metadata resource.
+
+        """
+
+        try:
+            return cls(
+                resource_name=resource_name,
+                metadata_store_id=metadata_store_id,
+                project=project,
+                location=location,
+                credentials=credentials,
+            )
+        except exceptions.NotFound:
+            logging.info(f"Resource {resource_name} not found.")
+
+    def update(
+        self,
+        metadata: Optional[Dict] = None,
```

Comment:
This should strictly be a `Dict` because passing Non would result in a TypeError on line 239.

Reply:
Gotcha. 
Yes, I was doing some ugly stuff in the _update_resource because of that...
`metadata=metadata if metadata else {}`, as it was failing marshalling.
Thanks!